.github/workflows/ci.yaml

```diff
@@ -64,304 +66,40 @@ jobs:
-        uses: actions-rs/toolchain@v1
-        with:
-          profile: minimal
-          toolchain: stable
-
-      - uses: subosito/flutter-action@v2
-        with:
-          channel: "stable"
-          flutter-version: '3.3.1'
-          architecture: x64
-
-      - uses: actions/setup-java@v1
-        with:
-          java-version: "8.x" # "betterprogramming.pub" says must be java "8"
-
-      - uses: actions/cache@v3
-        id: cache-deps
-        with:
-          path: |
-            ~/.cargo/bin/
-            ~/.cargo/registry/index/
-            ~/.cargo/registry/cache/
-            ~/.cargo/git/db/
-            ./frb_example/with_flutter/rust/target
-          key: ${{ runner.os }}-cargo-integrate-android-${{ hashFiles('**/Cargo.lock') }}
-
-      - name: Add Rust targets
-        run: rustup target add x86_64-linux-android
-
-      - name: Install `cargo-ndk`
-        if: steps.cache-deps.outputs.cache-hit != 'true'
-        run: cargo install cargo-ndk
-
-      - name: Install Flutter dependencies
-        working-directory: ./frb_example/with_flutter
-        run: flutter pub get
-
-      - name: Build Rust lib
-        working-directory: ./frb_example/with_flutter/rust
-        # NOTE since run on simulator, need -t to be x86_64. For real devices, of course arm-like ones.
-        run: cargo ndk -t x86_64 -o ../android/app/src/main/jniLibs build
-
-      - name: Run Flutter integration tests
-        continue-on-error: true
-        timeout-minutes: 20
-        uses: Wandalen/wretry.action@v1.0.25
-        with:
-          action: reactivecircus/android-emulator-runner@v2
-          with: |
-            api-level: 29
-            arch: x86_64
-            profile: ${{ matrix.device }}
-            script: cd frb_example/with_flutter && flutter test integration_test/main.dart --verbose
-
-      # # flaky GitHub Action testing environment... have to retry
-      # # see https://github.com/fzyzcjy/flutter_rust_bridge/issues/47
-      # - name: Run Flutter integration tests (retry)
-      #   id: run_flutter_integration_test_retry_1
-      #   if: steps.run_flutter_integration_test.outcome == 'failure'
-      #   continue-on-error: true
-      #   timeout-minutes: 30
-      #   uses: reactivecircus/android-emulator-runner@v2
-      #   with:
-      #     api-level: 29
-      #     arch: x86_64
-      #     profile: ${{ matrix.device }}
-      #     script: |
-      #       cd frb_example/with_flutter && flutter test integration_test/main.dart --verbose
-
-      # # https://github.community/t/how-to-retry-a-failed-step-in-github-actions-workflow/125880
-      # - name: Set status
-      #   if: always()
-      #   run: |
-      #     if ${{ steps.run_flutter_integration_test.outcome=='success' || steps.run_flutter_integration_test_retry_1.outcome=='success' }}; then
-      #        echo fine
-      #     else
-      #        exit 1
-      #     fi
-
+    uses: ./.github/workflows/flutter_android_test.yml
```

Comment:
nit: looks like we extract some parts out of separate files, and some parts still in ci.yaml. Is there a criterion which to extract or not? Or shall we just extract all things (e.g. dart web tests etc) to separate yml?

Reply:
actually we can do way better than that and refactor all the flutter_android_*.yml into a single reusable workflow (partly started it yesterday on a local branch), sequence workflows to use less CI time, etc but I leave it to another PR it's quite a bunch of additional work. @fzyzcjy 

> Is there a criterion which to extract or not? Or shall we just extract all things (e.g. dart web tests etc) to separate yml?

Well I'd say it mostly depends on the size of the workflow(s).

Reply:
I see. So shall I merge this now, or wait for the next?

Reply:
merge this now please 🙏 😆